src/terms.jl

```diff
@@ -116,18 +118,14 @@ julia> modelcols(f.rhs, (a=[3, 4], b=[4, 5]))
  2.302585092994046
 ```
 """
-struct FunctionTerm{Forig,Fanon,Names} <: AbstractTerm
-    forig::Forig
-    fanon::Fanon
+struct FunctionTerm{F,Args} <: AbstractTerm
+    f::F
+    args::Args
     exorig::Expr
-    args_parsed::Vector
 end
```

Comment:
Is it actually useful to parameterize on the function and argument types? Doesn't `broadcast` specialize on these anyway?

Reply:
we need the function at least for methods like `apply_schema` to handle specific syntax.  the args I could see the case for dropping sicne they go through the function barrier in broadcast, but I'd prefer to make that change uniformly for all teh term types to see what the overall runtime and compile time performance implications are